xdsl/dialects/riscv.py

```diff
@@ -331,6 +334,73 @@ class RISCVOp(Operation, ABC):
     def assembly_line(self) -> str | None:
         raise NotImplementedError()
 
+    @classmethod
+    def parse(cls, parser: Parser) -> Self:
+        operands = cls.parse_operands(parser)
+        successors = cls.parse_successors(parser)
+        regions = cls.parse_regions(parser)
+        attributes = cls.parse_attributes(parser)
+        parser.parse_punctuation(":")
+        result_types = cls.parse_function_type(parser).outputs.data
+        return cls.create(
+            operands=operands,
+            result_types=result_types,
+            attributes=attributes,
+            successors=successors,
+            regions=regions,
+        )
+
+    @classmethod
+    def parse_operands(cls, parser: Parser) -> Sequence[SSAValue]:
+        return parser.parse_comma_separated_list(
+            parser.Delimiter.PAREN, parser.parse_operand
+        )
+
+    @classmethod
+    def parse_successors(cls, parser: Parser) -> Sequence[Block]:
+        return parser.parse_optional_successors() or []
+
+    @classmethod
+    def parse_function_type(cls, parser: Parser) -> FunctionType:
+        return parser.parse_function_type()
+
+    @classmethod
+    def parse_attributes(cls, parser: Parser) -> Mapping[str, Attribute]:
+        return parser.parse_optional_attr_dict()
+
+    @classmethod
+    def parse_regions(cls, parser: Parser) -> Sequence[Region]:
+        if parser.parse_optional_punctuation("(") is not None:
+            regions = parser.parse_comma_separated_list(
+                parser.Delimiter.NONE, parser.parse_region
+            )
+            parser.parse_punctuation(")")
+            return regions
+        return []
+
+    def print(self, printer: Printer) -> None:
+        self.print_operands(printer)
+        printer.print_successors(self.successors)
+        printer.print_regions(self.regions)
+        self.print_attributes(printer)
+        printer.print(" : ")
+        self.print_types(printer)
+
+    def print_operands(self, printer: Printer) -> bool:
+        printer.print("(")
+        if self.operands:
+            printer.print_list(self.operands, printer.print_ssa_value)
+        printer.print(")")
```

Comment:
Let's just drop the brackets right away? No need to re-implement them in the custom syntax, no? And we are changing all the operations anyway.

Reply:
In this case, I see a merit to leave such changes for later. One can argue its easy, but there is a risk of getting into stylistic discussions which are typically faster if factored out. I feel exactly these "we can just sneak it in" habit leads to our PRs sometimes growing quite a bit.

Reply:
In particular, we then need to think about how instructions with immediates are handled, ... I feel keeping this discussion out of this PR will help us to move faster.

Reply:
Imo, removing the quotes from all operations, or removing quotes and brackets from all operations is roughly equivalent in terms of discussion required... But I can see the reason, let's not get hung up on this part.

Reply:
One is clearly minimal, the other is not. Also, I would actually not be too certain about this. Talking about operators, I see us already considering how to handle immediates. ... In any case, thank you for allowing us to focus first.